testing/run-scripts/run_cloud.sh

```diff
@@ -174,26 +191,29 @@ if ! docker ps -a | grep uproxy-sshd >/dev/null; then
         echo -n "$BANNER" > $TMP_DIR/banner
         echo -n "$PUBLIC_IP" > $TMP_DIR/hostname
 
-        # Optional build args aren't very flexible...confine the messiness here.
-        ISSUE_INVITE_ARGS="-c"
-        if [ -n "$INVITE_CODE" ]
-        then
-            ISSUE_INVITE_ARGS="$ISSUE_INVITE_ARGS -i $INVITE_CODE"
-        fi
-        docker build --build-arg issue_invite_args="$ISSUE_INVITE_ARGS" -t uproxy/sshd $TMP_DIR
+        docker build -t uproxy/sshd $TMP_DIR
     fi
 
     # Add an /etc/hosts entry to the Zork container.
     # Because the Zork container runs with --net=host, we can't use the
     # regular, ever-so-slightly-more-elegant Docker notation.
-    HOST_IP=`ip -o -4 addr list docker0 | awk '{print $4}' | cut -d/ -f1`
     docker run --restart=always -d -p $SSHD_PORT:22 --name uproxy-sshd --add-host zork:$HOST_IP uproxy/sshd > /dev/null
 
-    echo -n "Waiting for Zork to come up..."
-    while ! ((echo ping ; sleep 0.5) | nc -w 1 $HOST_IP 9000 | grep ping) > /dev/null; do echo -n .; done
-    echo "ready!"
+    # Configure access.
+    # In descending order of preference:
+    #  - migrate authorized_keys or apply -i
+    #  - issue a new invite
+    if [ -n "$GIVER_AUTH_KEYS" ] || [ -n "$GETTER_AUTH_KEYS" ]
+    then
+        docker exec uproxy-sshd sh -c "echo $GIVER_AUTH_KEYS > /home/giver/.ssh/authorized_keys"
+        docker exec uproxy-sshd sh -c "echo $GETTER_AUTH_KEYS > /home/getter/.ssh/authorized_keys"
+    else
+        if [ -n "$INVITE_CODE" ]
+        then
+            INVITE_CODE=`docker exec uproxy-sshd /issue_invite.sh -i "$INVITE_CODE"`
+        else
+            INVITE_CODE=`docker exec uproxy-sshd /issue_invite.sh -c -u giver`
+            echo -e "\nINVITE CODE URL:\n$INVITE_CODE"
```

Comment:
Before we were echoing the invite code even if sshd was already started - any particular reason for the change?